sdk/src/Core/Amazon.Runtime/ClientConfig.cs

```diff
@@ -832,6 +832,31 @@ public TimeSpan? Timeout
             }
         }
 
+#if NET8_0_OR_GREATER
+        TimeSpan? _connectTimeout;
+
+        /// <summary>
+        /// Gets and sets the connection timeout that will be set on the HttpClient used by the service client to make requests.
+        /// The connection timeout is used control the wait time for the connection to be established to the service. The default
+        /// connection timeout for the HttpClient is infinite waiting period.
+        /// </summary>
+        public TimeSpan? ConnectTimeout
+        {
+            get
+            {
+                if (!this._connectTimeout.HasValue)
+                    return null;
```

Comment:
Ideally I would like to default this to `DefaultConfiguration.ConnectTimeout` instead of null but changing the default timeout would be a breaking change. Maybe when we do a 4.1 we can reconsider the default.